soql-standalone-parser/src/main/scala/com/socrata/soql/ast/Expression.scala

```diff
@@ -171,200 +161,147 @@ object SpecialFunctions {
-            sb.append("NOT ")
-            parameters(0).format(d, sb, limit)
+            d"NOT ${parameters(0).doc}"
           case _ =>
-            sb.append(op)
-            parameters(0).format(d, sb, limit)
+            d"${op}${parameters(0).doc}"
         }
       case SpecialFunctions.Operator(op) if parameters.size == 2 =>
-        for {
-          sb <- parameters(0).format(d, sb, limit)
-          _ = sb.append(" ").append(op).append(" ")
-          sb <- parameters(1).format(d, sb, limit)
-        } yield sb
+        op match {
+          case "AND" | "OR" | "+" | "*" =>
+            // We're going to format this like
+            //    blah
+            //      OP blah
+            //      OP blah
+            // taking advantage of the operator's associativity
+            val builder = new VectorBuilder[Expression]
+            variadizeAssociative(builder)
+            val result = builder.result()
+            (maybeParens(result.head) +: result.drop(1).map { e => Doc(op) +#+ maybeParens(e) }).sep.hang(2)
+          case _ =>
+            Seq(maybeParens(parameters(0)), Doc(op) +#+ maybeParens(parameters(1))).sep.hang(2)
+        }
       case SpecialFunctions.Operator(op) =>
         sys.error("Found a non-unary, non-binary operator: " + op + " at " + position)
       case SpecialFunctions.Cast(typ) if parameters.size == 1 =>
-        parameters(0).format(d, sb, limit).map(_.append(" :: ").append(typ))
-      case SpecialFunctions.Cast(_) =>
-        sys.error("Found a non-unary cast at " + position)
+        d"${parameters(0).doc} :: ${typ.toString}"
       case SpecialFunctions.Between =>
-        for {
-          sb <- parameters(0).format(d, sb, limit)
-          _ = sb.append(" BETWEEN ")
-          sb <- parameters(1).format(d, sb, limit)
-          _ = sb.append(" AND ")
-          sb <- parameters(2).format(d, sb, limit)
-        } yield sb
+        Seq(parameters(0).doc, d"BETWEEN ${parameters(1).doc}", d"AND ${parameters(2).doc}").sep.hang(2)
       case SpecialFunctions.NotBetween =>
-        for {
-          sb <- parameters(0).format(d, sb, limit)
-          _ = sb.append(" NOT BETWEEN ")
-          sb <- parameters(1).format(d, sb, limit)
-          _ = sb.append(" AND ")
-          sb <- parameters(2).format(d, sb, limit)
-        } yield sb
+        Seq(parameters(0).doc, d"NOT BETWEEN ${parameters(1).doc}", d"AND ${parameters(2).doc}").sep.hang(2)
       case SpecialFunctions.IsNull =>
-        parameters(0).format(d, sb, limit).map(_.append(" IS NULL"))
+        d"${parameters(0).doc} IS NULL"
       case SpecialFunctions.IsNotNull =>
-        parameters(0).format(d, sb, limit).map(_.append(" IS NOT NULL"))
+        d"${parameters(0).doc} IS NOT NULL"
       case SpecialFunctions.In =>
-        parameters(0).format(d, sb, limit).map(_.append(" IN (")).flatMap { sb =>
-          appendParams(sb, parameters.iterator.drop(1), ",", d, limit)
-        }.map(_.append(")"))
-      case SpecialFunctions.NotIn =>
```

Comment:
is NotIN not an available function? Maybe it gets interpreted as a NOT and then an IN?

Reply:
Er, nope, I just missed it.